src/basic/log.c

```diff
@@ -1689,6 +1689,39 @@ int log_syntax_invalid_utf8_internal(
                                    "String is not UTF-8 clean, ignoring assignment: %s", strna(p));
 }
 
+int log_syntax_parse_error_internal(
+                const char *unit,
+                const char *config_file,
+                unsigned config_line,
+                int error,
+                bool critical,
+                const char *file,
+                int line,
+                const char *func,
+                const char *lvalue,
+                const char *rvalue) {
+
+        _cleanup_free_ char *escaped = NULL;
+
+        if (rvalue && !utf8_is_valid(rvalue)) {
+                escaped = utf8_escape_invalid(rvalue);
+                if (!escaped)
+                        rvalue = "(oom)";
+                else
+                        rvalue = "(escaped)";
+        }
+
+        log_syntax_internal(unit, critical ? LOG_ERR : LOG_WARNING, config_file, config_line, error,
+                            file, line, func,
+                            "Failed to parse %s=%s%s%s%s%s",
+                            strna(lvalue), strempty(rvalue), strempty(escaped),
```

Comment:
so you output the unescaped rvalue? wouldn't it make more sense to just output the escaped version?